deepspeed/runtime/zero/linear.py

```diff
@@ -29,8 +30,13 @@ def print_rank_0(message, debug=False, force=False):
 
 
 try:
-    autocast_custom_fwd = torch.cuda.amp.custom_fwd
-    autocast_custom_bwd = torch.cuda.amp.custom_bwd
+    device = get_accelerator().device_name()
```

Comment:
I think it might be better to push the `try ... except` only into the `cuda` code path.

```python

if device == 'cuda':
   try:
     ...
   except:
     ...
else:
     autocast_custom_fwd = noop_decorator
     autocast_custom_bwd = noop_decorator
```

Reply:
Good suggestion, code updated.